taker-frontend/src/useEventSource.ts

```diff
@@ -12,11 +12,16 @@ export function useEventSource(url: string, withCredentials?: boolean) {
 
         es.addEventListener("error", () => {
             setIsConnected(false);
+            // we call set source here which changes the state of this component causing a re-render
+            // as far as i understand this will be called everytime an event is sent from the backend
+            // have not investigated whether the backend is sending this event multiple times
```

Comment:
Calling `setSource` shouldn't trigger a re-render AFAIK, because `source` isn't part of the dependencies of this hook.

Reply:
I think it will trigger a re-render outside of this hook for everything that depends on source!

Reply:
That's true, but it was also intended so that we keep checking for heartbeats after we get an `EventSource`.

There's probably a better way of doing all this and I'm not surprised we've encountered a bug, but the logic still makes sense to me.

Reply:
Yeah this logic also makes sense to me!